match/Repetitions.roc

```diff
@@ -0,0 +1,80 @@
+interface Repetitions
+    exposes [
+        match,
+    ]
+    imports []
+
+match : Str, Str -> Bool
+match = \pattern, text ->
+    # convert pattern and text from Str to List U8
+    patternList = Str.toUtf8 pattern
+    textList = Str.toUtf8 text
+
+    # if the pattern begins with a ^, we'll only check for matches at the beginning of `text`
+    if List.first patternList == Ok '^' then
+        matchHere (List.dropFirst patternList 1) textList
+    else
+        # we'll always need to consider at least 0 as a starting point since text may be empty
+        startingPoints = List.range { start: At 0, end: At (Num.max 1 (List.len textList)) }
+
+        # find the first starting point in `text` that `pattern` matches.
+        startingPoints
+        |> List.findFirst \index -> matchHere patternList (List.dropFirst textList index)
+        |> Result.isOk
+
+matchHere = \pattern, text ->
+    when (pattern, text) is
+        ([], _) -> Bool.true
+        (['$'], []) -> Bool.true
+        # match any character followed by a *
+        ([p, '*', ..], _) -> matchStar p (List.dropFirst pattern 2) text
+        ([p, ..], [t, ..]) if p == t || p == '.' ->
+            matchHere (List.dropFirst pattern 1) (List.dropFirst text 1)
```

Comment:
Using rest patterns with list would clean things up slightly
```suggestion
        ([p, .. as ps], [t, .. as ts]) if p == t || p == '.' ->
            matchHere ps ts
```

Reply:
Agree, but for the time being it's triggering [this issue](https://github.com/roc-lang/roc/issues/6606)

Reply:
Ah, bummer